logic/pulsed/sequence_generator_logic.py

```diff
@@ -2071,3 +2238,169 @@ def _delete_sequence(self, names):
                 self.pulsegenerator().delete_sequence(seq)
         self.sigAvailableSequencesUpdated.emit(self.sampled_sequences)
         return
+
+    @QtCore.Slot()
+    def run_pg_benchmark(self, t_goal=10):
+
+        def round_to_granularity(n_samples):
+            granularity = self.pulsegenerator().get_constraints().waveform_length.step
+            return np.ceil(n_samples / granularity) * granularity
+
+        self._benchmark_write.reset()
+        self._benchmark_load.reset()
+
+        waveform_name = 'qudi_benchmark_chunk'
+        n_init_guess = [1e4, 1e6]
+        n_max_fix = 10e6
```

Comment:
These hard-coded parameters limit the generality. Open for suggestions, but hard to guess reasonable benchmark chunk sizes without knowing the coarse speed of the device.